libkbfs/folder_block_ops.go

```diff
@@ -941,6 +971,20 @@ func (fbo *folderBlockOps) newDirDataWithLBC(lState *lockState,
 		}, fbo.log)
 }
 
+func (fbo *folderBlockOps) makeDirDirtyLocked(
+	lState *lockState, ptr BlockPointer, unrefs []BlockInfo) func() {
+	fbo.blockLock.AssertLocked(lState)
+	oldUnrefs, wasDirty := fbo.dirtyDirs[ptr]
+	fbo.dirtyDirs[ptr] = append(oldUnrefs, unrefs...)
+	return func() {
+		if wasDirty {
+			fbo.dirtyDirs[ptr] = oldUnrefs[:len(oldUnrefs)]
```

Comment:
Did you mean to do `fbo.dirtyDirs[ptr] = oldUnrefs[:len(oldUnrefs):len(oldUnrefs)]` here? It seems the length of `oldUnrefs` never changed.

Reply:
This is an undo function, so it's restoring the state before `makeDirDirtyLocked` was applied.

Reply:
Yeah I was just worried that the above `append` would have changed the length of `oldUnrefs`, and I was trying to get exactly the original slice.  Does the double-colon syntax help do that, I've never seen it before.  I can try it.

Reply:
Looks like the third index sets the capacity of the slice. Seems redundant in this situation.
I wonder if you can just do `oldUnrefs[:]` here though.

Reply:
The double colon limits the underlying capacity, to make sure an `append` at a later time can't override underlying data. So for example, if you have

```
a := []int{1, 2, 3}
b := a[:2]
c := a[:2:2]
```

Then `d := append(b, 1)` can override the third element in `a`, but `d := append(c, 1)` can't: https://play.golang.org/p/xACLZaiHrMk

If all you care is the length, then just returning `oldUnrefs` would be the same right?

Reply:
Wow, TIL! That definitely breaks an assumption I had about how slices work.

Reply:
I used to get confused about how slice works, but this showed up and explained it very well: https://blog.golang.org/go-slices-usage-and-internals

Reply:
Cool thanks, good explanation!  You're right it can probably just be `unrefs`, but it doesn't seem worth changing at this point and this way is still a little clearer to me.

Reply:
Yeah; not suggesting a change; Just a discussion. What you have now looks great to me!